pyi.py

```diff
@@ -246,6 +246,31 @@ def _check_platform_check(self, node: ast.Compare) -> None:
         else:
             self.error(node, Y007)
 
+    def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
+        self.generic_visit(node)
+
+        for i, statement in enumerate(node.body):
+            if i == 0:
+                # normally, should just be "..."
+                if isinstance(statement, ast.Pass):
+                    # should disable this if https://github.com/python/typeshed/pull/1009 isn't
+                    # accepted
+                    self.error(statement, Y009)
+                    continue
+                elif isinstance(statement, ast.Expr) and isinstance(statement.value, ast.Ellipsis):
+                    continue
+            # allow "raise", a number of stubs have this
+            if isinstance(statement, ast.Raise):
+                continue
+            # allow assignments in constructor for now
+            # (though these should probably be changed)
+            if node.name == '__init__' and isinstance(statement, ast.Assign) and \
+                    isinstance(statement.targets[0], ast.Attribute) and \
+                    isinstance(statement.targets[0].value, ast.Name) and \
+                    statement.targets[0].value.id == 'self':
+                continue
```

Comment:
You can check for type comments by getting `lines` passed from the checker to the Visitor and looking at the corresponding line.

But, I actually agree with you that it's not worth having an elaborate case for this. Simple is better than complex. To not shoot ourselves in the foot with adoption, I'd simply ignore the Y010 check entirely for `__init__` methods. Much less code and no false positives.